### ISSUE TITLE
Creating Revision.......interrupted

### ISSUE THREAD

**Issue by selfagency** [google-github-actions/deploy-cloudrun#382]
### TL;DR

Deployments of container images to Cloud Run fail inexplicably.

### Expected behavior

For deployments to succeed or to at least provide a reason they're failing.

### Observed behavior

Deployments fail with the message "Creating Revision.......interrupted".

### Action YAML

```yaml
name: ↳ Deployment subworkflow
on:
  workflow_call:
    inputs:
      environment:
        description: The environment to deploy to
        type: string
        required: true
      project:
        description: Google Cloud project
        type: string
        required: true
      service:
        description: The service to deploy
        type: string
        required: true
      sha:
        description: The SHA of the commit to deploy
        type: string
        required: false
      promote:
        description: Whether to promote the new version
        type: string
        default: true
        required: false
      skip:
        description: Whether to skip the deployment
        type: string
        default: false
        required: false
      version:
        description: Deployment version tag
        type: string
        required: false
      filestore-ip:
        description: Filestore IP address for ephemeral deployment persistent storage
        type: string
        required: false
      filestore-name:
        description: Filestore name for ephemeral deployment persistent storage
        type: string
        required: false
      options:
        description: Container options
        type: string
        required: false
      port:
        description: Container port
        type: string
        required: false
      env_vars:
        description: Container environment variables
        type: string
        required: false
    secrets:
      WORKLOAD_ID_PROVIDER_STAGING:
        description: The workload ID provider for staging
        required: true
      SERVICE_ACCOUNT_STAGING:
        description: The service account for staging
        required: true
      WORKLOAD_ID_PROVIDER_PRODUCTION:
        description: The workload ID provider for production
        required: false
      SERVICE_ACCOUNT_PRODUCTION:
        description: The service account for production
        required: false
      RELIABILITY_SSH_KEY:
        description: Reliability SSH key
        required: true
      UD_PAT:
        description: UD Personal Access Token
        required: true
    outputs:
      status:
        description: Whether deployment succeeded or failed
        value: ${{ jobs.deploy.outputs.status }}
env:
  DEPLOYMENT: ${{ inputs.environment }}
  PROJECT: ${{ inputs.project }}
  SERVICE: ${{ inputs.service }}
  SHA: ${{ inputs.sha }}
jobs:
  deploy:
    name: ${{ inputs.environment }}
    runs-on: ubuntu-latest
    timeout-minutes: 30
    permissions:
      id-token: write
      contents: read
    outputs:
      status: ${{ steps.validation.outputs.status }}
    steps:
      - name: Checkout repo
        uses: 'actions/checkout@v3'
        if: inputs.skip != 'true'
        with:
          ssh-key: ${{ secrets.RELIABILITY_SSH_KEY }}
          token: ${{ secrets.UD_PAT }}

      - name: Get PR number
        id: pr-number
        if: ${{ inputs.environment == 'ephemeral' }}
        uses: 8BitJonny/gh-get-current-pr@2.0.0
        with:
          github-token: ${{ secrets.UD_PAT }}

      - name: Set deployment parameters
        id: params
        if: inputs.skip != 'true'
        uses: 'actions/github-script@v6'
        env:
          PR: ${{ steps.pr-number.outputs.number }}
        with:
          script: require('./ci/deploy/params.js')({core});

      - name: Authenticate to GCP `unstoppable-domains` project
        uses: 'google-github-actions/auth@v0'
        if: inputs.environment == 'production' && inputs.skip != 'true'
        with:
          workload_identity_provider: ${{ secrets.WORKLOAD_ID_PROVIDER_PRODUCTION }}
          service_account: ${{ secrets.SERVICE_ACCOUNT_PRODUCTION }}

      - name: Authenticate to GCP `unstoppable-domains-staging` project
        uses: 'google-github-actions/auth@v0'
        if: inputs.environment != 'production' && inputs.skip != 'true'
        with:
          workload_identity_provider: ${{ secrets.WORKLOAD_ID_PROVIDER_STAGING }}
          service_account: ${{ secrets.SERVICE_ACCOUNT_STAGING }}

      - name: Deploy ${{ inputs.service }} to App Engine ${{ inputs.environment }}
        uses: 'google-github-actions/deploy-appengine@v0'
        id: deploy-gae
        if: inputs.skip != 'true' && inputs.environment != 'sandbox' && inputs.environment != 'ephemeral'
        continue-on-error: true
        with:
          working_directory: ${{ github.workspace }}
          project_id: ${{ steps.params.outputs.project }}
          deliverables: ${{ steps.params.outputs.deliverables }}
          image_url: ${{ steps.params.outputs.image_url }}
          version: ${{ inputs.version }}
          promote: false

      - name: Deploy ${{ inputs.service }} to Cloud Run ${{ inputs.environment }}
        uses: 'google-github-actions/deploy-cloudrun@v0'
        id: deploy-gcr
        if: inputs.skip != 'true' && inputs.environment != 'production' && inputs.environment != 'staging'
        continue-on-error: true
        with:
          service: ${{ steps.params.outputs.service }}
          env_vars: ${{ steps.params.outputs.env_vars }}
          image: ${{ steps.params.outputs.image_url }}
          flags: ${{ steps.params.outputs.flags }}
          no_traffic: ${{ inputs.environment != 'ephemeral' }}

      - name: Validate deployment succeeded
        id: validation
        if: always()
        uses: 'actions/github-script@v6'
        with:
          script: |
            const gaeOutcome = '${{ steps.deploy-gae.outcome }}';
            const gcrOutcome = '${{ steps.deploy-gcr.outcome }}';
            const status = '${{ inputs.skip }}' != 'true' && (gaeOutcome !== 'success' && gcrOutcome !== 'success') ? 'failure' : 'success';
            core.setOutput('status', status);
```

### GitHub log output

```text
##[debug]Caching tool gcloud 398.0.0 x64
##[debug]source dir: /home/runner/work/_temp/92dc672e-5f56-4d02-8833-dfa5f0b0ec51/google-cloud-sdk
##[debug]destination /opt/hostedtoolcache/gcloud/398.0.0/x64
##[debug]finished caching tool
##[debug]Running command: gcloud --quiet auth login --cred-file /home/runner/work/unstoppable-domains-website/unstoppable-domains-website/gha-creds-a592a5a3f7f2a4c8.json
##[debug]Running command: gcloud auth list
Running: gcloud run deploy pr-5503-cache --image gcr.io/unstoppable-domains-staging/ud-redis:latest --quiet --platform managed --region us-central1 --cpu 2000m --ingress internal --max-instances 1 --memory 4Gi --min-instances 0 --port 6379 --service-account run-cloud@unstoppable-domains-staging.iam.gserviceaccount.com --vpc-connector vpc-serverless --project unstoppable-domains-staging --format json
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy pr-5503-cache --image gcr.io/unstoppable-domains-staging/ud-redis:latest --quiet --platform managed --region us-central1 --cpu 2000m --ingress internal --max-instances 1 --memory 4Gi --min-instances 0 --port 6379 --service-account run-cloud@unstoppable-domains-staging.iam.gserviceaccount.com --vpc-connector vpc-serverless --project unstoppable-domains-staging --format json`: Deploying container to Cloud Run service [pr-5503-cache] in project [unstoppable-domains-staging] region [us-central1]
Deploying...
Creating Revision.......interrupted
Deployment failed
ERROR: (gcloud.run.deploy) Revision 'pr-5503-cache-00009-nak' is not ready and cannot serve traffic.
##[debug]Node Action run completed with exit code 1
##[debug]CLOUDSDK_METRICS_ENVIRONMENT='github-actions-deploy-cloudrun'
##[debug]Finishing: Deploy cache to Cloud Run ephemeral
```

### GCP log output

```text
Notice 2022-08-19T00:02:27.007548Z Cloud Run ReplaceService pr-5503-cache githubactions@unstoppable-domains-staging.iam.gserviceaccount.com {@type: type.googleapis.com/google.cloud.audit.AuditLog, authenticationInfo: {…}, authorizationInfo: […], methodName: google.cloud.run.v1.Services.ReplaceService, request: {…}, requestMetadata: {…}, resourceLocation: {…}, resourceName: namespaces/unstoppable-domains-staging/services/pr-5503-cache, r…
Error 2022-08-19T00:02:36.526472Z Cloud Run pr-5503-cache-00011-juy {@type: type.googleapis.com/google.cloud.audit.AuditLog, resourceName: namespaces/unstoppable-domains-staging/revisions/pr-5503-cache-00011-juy, response: {…}, serviceName: run.googleapis.com, status: {…}}
Error 2022-08-19T00:02:36.694566016Z terminated: Application failed to start: Failed to create init process: failed to load /usr/local/bin/docker-entrypoint.sh: exec format error
Warning 2022-08-19T00:02:36.751483519Z Application exec likely failed
```

### Additional information

I don't know if it has anything to do with what's happening but I am trying to use Cloud Run with Filestore as my persistent storage to run temporary Postgres and Redis instances for application testing purposes. I am following the directions [here](https://cloud.google.com/run/docs/tutorials/network-filesystems-filestore) and it notes that the gen 2 execution environment is required. I couldn't figure out how to force it to use that version using this action nor, as I said, am I sure it's even relevant.

**Comment by selfagency** [comment]
I changed to a full path for the `vpc-connector` flag and then it didn't interrupt, but it still failed without a cause and with the same GCP log output as above.

```
##[debug]Caching tool gcloud 398.0.0 x64
##[debug]source dir: /home/runner/work/_temp/8ccf2629-0e0a-4d3d-9cc9-a2f3ed4f181e/google-cloud-sdk
##[debug]destination /opt/hostedtoolcache/gcloud/398.0.0/x64
##[debug]finished caching tool
##[debug]Running command: gcloud --quiet auth login --cred-file /home/runner/work/unstoppable-domains-website/unstoppable-domains-website/gha-creds-6542d6211a14fc7a.json
##[debug]Running command: gcloud auth list
Running: gcloud run deploy pr-5503-cache --image gcr.io/unstoppable-domains-staging/ud-redis:latest --quiet --platform managed --region us-central1 --cpu 2000m --ingress internal --max-instances 1 --memory 4Gi --min-instances 0 --port 6379 --service-account run-cloud@unstoppable-domains-staging.iam.gserviceaccount.com --vpc-connector projects/unstoppable-domains-staging/locations/us-central1/connectors/vpc-serverless --project unstoppable-domains-staging --format json
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy pr-5503-cache --image gcr.io/unstoppable-domains-staging/ud-redis:latest --quiet --platform managed --region us-central1 --cpu 2000m --ingress internal --max-instances 1 --memory 4Gi --min-instances 0 --port 6379 --service-account run-cloud@unstoppable-domains-staging.iam.gserviceaccount.com --vpc-connector projects/unstoppable-domains-staging/locations/us-central1/connectors/vpc-serverless --project unstoppable-domains-staging --format json`: Deploying container to Cloud Run service [pr-5503-cache] in project [unstoppable-domains-staging] region [us-central1]
Deploying...
Creating Revision........interrupted
Deployment failed
ERROR: (gcloud.run.deploy) Cloud Run error: The user-provided container failed to start and listen on the port defined provided by the PORT=6379 environment variable. Logs for this revision might contain more information.
```

**Comment by selfagency** [comment]
My Postgres container does something different but has the same GCP log output:

```
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy pr-5503-db --image gcr.io/unstoppable-domains-staging/ud-postgres:latest --quiet --platform managed --region us-central1 --update-env-vars POSTGRES_DB="ud-staging" --cpu 2000m --ingress internal --max-instances 1 --memory 4Gi --min-instances 0 --port 5432 --service-account run-cloud@unstoppable-domains-staging.iam.gserviceaccount.com --vpc-connector projects/unstoppable-domains-staging/locations/us-central1/connectors/vpc-serverless --project unstoppable-domains-staging --format json`: Deploying container to Cloud Run service [pr-5503-db] in project [unstoppable-domains-staging] region [us-central1]
Deploying...
Creating Revision.........................................................................................................failed
Deployment failed
ERROR: (gcloud.run.deploy) Cloud Run error: The user-provided container failed to start and listen on the port defined provided by the PORT=5432 environment variable. Logs for this revision might contain more information.
```

```
Notice 2022-08-19T01:05:59.038310Z Cloud Run ReplaceService pr-5503-dbgithubactions@unstoppable-domains-staging.iam.gserviceaccount.com {@type: type.googleapis.com/google.cloud.audit.AuditLog, authenticationInfo: {…}, authorizationInfo: […], methodName: google.cloud.run.v1.Services.ReplaceService, request: {…}, requestMetadata: {…}, resourceLocation: {…}, resourceName: namespaces/unstoppable-domains-staging/services/pr-5503-db, resp…
Error 2022-08-19T01:06:07.524504Z Cloud Run pr-5503-db-00011-cod {@type: type.googleapis.com/google.cloud.audit.AuditLog, resourceName: namespaces/unstoppable-domains-staging/revisions/pr-5503-db-00011-cod, response: {…}, serviceName: run.googleapis.com, status: {…}}
Error 2022-08-19T01:06:07.759487378Z terminated: Application failed to start: Failed to create init process: failed to load /usr/local/bin/docker-entrypoint.sh: exec format error
Warning 2022-08-19T01:06:07.818258139Z Application exec likely failed
```

**Comment by selfagency** [comment]
Here's the Dockerfile

```
FROM redis:5.0

RUN apt-get update -y && apt-get install -y \
    nfs-common \
    sudo \
    && apt-get clean

COPY . .

ENV REDISDATA /mnt/filestore/redis/pr-${PR_NUMBER}

RUN useradd -m docker && echo "docker:docker" | chpasswd && adduser docker sudo
RUN echo "docker ALL=(ALL) NOPASSWD: ALL" >>/etc/sudoers.d/docker
RUN mkdir -p /mnt/filestore
RUN chown docker /mnt/filestore
USER docker

EXPOSE 6379
ENTRYPOINT ["run.sh"]
```

and the run script

```
#!/bin/bash
set -eo pipefail
sudo mkdir -p /filestore
sudo mount -o nolock "10.49.32.26:/ephemeral_data" "/mnt/filestore"
rm -rf /data
ln -s "${REDIS_DATA}" /data
redis-server --save 60 1 --loglevel warning
```

**Comment by selfagency** [comment]
Manually redeploying with v2 did not help

**Comment by selfagency** [comment]
It was because I was building my container locally on an M1 Mac. Building on Cloud Build resolved.
